### PR TITLE
view.topics;

### PR DESCRIPTION
**Pull request by pz-2015**
1). added view.topics support;
2) .topics: {'ssePath': true/'coopEventOrCallbackFnName'/fn(data, coopEvent)/{'server-event': fn(data)}}

### CODE REVIEW

**Comment by pz-2015** [comment]
1). modified app.sse and server.topic per discussion.
2). aligned app.poll, view.pollings with app.sse/app.ws and view.topics/view.channels;
3). included macha and chai for writing test case later;

TODO: write test case for app.sse using mocha and chai...

**Comment by pz-2015** [comment]
added test for app.sse and view.topics;
fixed bug in view.topics;
fixed bug in SSE implementation at backend;

**Comment by pz-2015** [comment]
!!Caveat: cleanup the test results in mocha.suite.suites. otherwise same test will be load multiple times on context switching!!

File: implementation/js/src/infrastructure/api.js [hunk 1]
```diff
@@ -411,11 +411,16 @@
...
 		    //stop everything
-		    if (url === false)
-		        return _.map(this._polls, function(card) {
+		    if (url === false){
+		    	return _.map(this._polls, function(card) {
 		            return card.cancel();
 		        });
+		    }else{
+		    	//align with app.sse and app.ws, there is suppose to be a default url defined in app.config.defaultPollingUrl
+		    	url = url || app.config.defaultPollingUrl || '/poll';
```
Comment (bluekvirus) [inline implementation/js/src/infrastructure/api.js]: @mr-beaver What's an example of the defaultPollingUrl?

Comment (pz-2015) [reply implementation/js/src/infrastructure/api.js]: app.config.defaultPollingUrl could take reference to '/sample/sse', which is an example of topic subscribing.

Comment (bluekvirus) [reply implementation/js/src/infrastructure/api.js]: defaultWS and SSE are ok, URL should be different for each kinda data.

**Pull request merged**